pkg/upload/xcode-build.go

```diff
@@ -2,159 +2,134 @@ package upload
-
-				// Check if dsymPath exists before proceeding
-				_, err := os.Stat(possibleDsymPath)
-				if err == nil {
+				if _, err = os.Stat(possibleDsymPath); err == nil {
 					dsymPath = possibleDsymPath
 					logger.Debug(fmt.Sprintf("Using dSYM path: %s", dsymPath))
 				}
 			}
 		}
 
+		// Default project root to current directory if not set
 		if dsymOptions.ProjectRoot == "" {
 			dsymOptions.ProjectRoot, _ = os.Getwd()
 			logger.Info(fmt.Sprintf("Setting `--project-root` to current working directory: %s", dsymOptions.ProjectRoot))
 		}
 
+		// Validate dSYM path
 		if dsymPath == "" {
-			return fmt.Errorf("No dSYM locations detected. Please provide a valid dSYM path or an Xcode project/workspace path")
+			return fmt.Errorf("No dSYM locations detected. Provide a valid dSYM path or Xcode project/workspace path")
 		}
 
+		// Locate and process dSYM files
 		dwarfInfo, tempDir, err = ios.FindDsymsInPath(dsymPath, dsymOptions.IgnoreEmptyDsym, dsymOptions.IgnoreMissingDwarf, logger)
 		tempDirs = append(tempDirs, tempDir)
 		if err != nil {
-			return err
-		} else if len(dwarfInfo) == 0 {
-			return fmt.Errorf("No dSYM files found in: %s", dsymPath)
+			return fmt.Errorf("Error locating dSYM files: %w", err)
 		}
-
-		// If the Info.plist path is not defined, we need to build the path to Info.plist from build settings values
-		if plistPath == "" && options.ApiKey == "" {
-			if buildSettings != nil {
-				plistPathExpected := filepath.Join(buildSettings.ConfigurationBuildDir, buildSettings.InfoPlistPath)
-				if utils.FileExists(plistPathExpected) {
-					plistPath = plistPathExpected
-					logger.Debug(fmt.Sprintf("Found Info.plist at expected location: %s", plistPath))
-				} else {
-					logger.Debug(fmt.Sprintf("No Info.plist found at expected location: %s", plistPathExpected))
-				}
-			}
+		if len(dwarfInfo) == 0 {
+			return fmt.Errorf("No dSYM files found in: %s", dsymPath)
 		}
 
-		// If the Info.plist path is defined and we still don't know the apiKey try to extract them from it
-		if plistPath != "" && options.ApiKey == "" {
-			// Read data from the plist
-			plistData, err = ios.GetPlistData(plistPath)
-			if err != nil {
-				return err
-			}
-
-			if options.ApiKey == "" {
-				options.ApiKey = plistData.BugsnagProjectDetails.ApiKey
-				if options.ApiKey != "" {
-					logger.Debug(fmt.Sprintf("Using API key from Info.plist: %s", options.ApiKey))
-				}
+		// Locate Info.plist if not already specified
+		if plistPath == "" && options.ApiKey == "" && buildSettings != nil {
+			plistPathExpected := filepath.Join(buildSettings.ConfigurationBuildDir, buildSettings.InfoPlistPath)
+			if utils.FileExists(plistPathExpected) {
+				plistPath = plistPathExpected
+				logger.Debug(fmt.Sprintf("Found Info.plist at: %s", plistPath))
+			} else {
+				logger.Debug(fmt.Sprintf("Info.plist not found at: %s", plistPathExpected))
 			}
 		}
```

Comment:
Can `ProcessDsymUpload` do the plist path exists check, so you just need to pass in where you think it is?

Reply:
It looks like ProcessDsymUpload is already doing a `FileExists` check so this is not needed.